lockfile/src/cyclonedx.rs

```diff
@@ -164,23 +164,37 @@ fn from_purl<T: Component>(component: &T) -> anyhow::Result<Package> {
 
 pub struct CycloneDX;
 
+impl CycloneDX {
+    fn process_components<T: Component>(components: Option<&[T]>) -> anyhow::Result<Vec<Package>> {
+        components.map_or(Ok(Vec::new()), |comp| {
+            let component_iter = filter_components(comp);
+            let mut packages = Vec::new();
+            for comp in component_iter {
+                match from_purl(comp) {
+                    Ok(pkg) => packages.push(pkg),
+                    Err(e) => {
+                        if e.is::<UnknownEcosystem>() {
+                            log::warn!("{:?}", e)
```

Comment:
I think for the GitHub app this is almost okay (if we even process CycloneDX files there). When the GitHub app processes lockfiles, there is a whitelist of log categories which are collected into the check run output and this could be added to inform the user when an unknown ecosystem is encountered.

However, I think it really should be logging using normal printing, not debug printing, if we are going to do something like that. It's probably the same for CLI usage.

It'd probably be cleaner if either this function had a callback that it uses for filtering errors and that callback was responsible for logging if necessary, or if the model allowed returning things that are maybe understood and maybe not.

There is an option in the PURL library to use `GenericPurl<String>` to parse any ecosystem, but ecosystem-specific rules aren't applied in that case, which can lead to invalid PURLs being accepted or non-canonical PURLs not being normalized. That could possibly used for returning information for unknown ecosystems if we go that route.

Reply:
It would be nice if the message provided more information, like which `package_type` it does not recognize and a notice about how that package will be skipped (maybe also include the `purl` entry).

Reply:
To add a bit of history about logging in parsers, see 8c5fe5f where we downgraded warnings to debug messages.

That issue was related to the generic lockfile parser (aka [`try_get_packages()`][1]). We have made efforts to reduce the use of that function, but it still exists and creates annoying issues such as this.

[1]: https://github.com/phylum-dev/cli/blob/50cb0d486ab75a8371f67c31178bbc1311850b05/cli/src/commands/parse.rs#L198-L215

I realize that this code was probably just copied from [the spdx parser][2], but I think this should either be downgraded to `log::debug!` or be removed completely. We can and should consider the alternatives to warning suggested above, but they are outside the scope of the PR.

[2]: https://github.com/phylum-dev/cli/blob/50cb0d486ab75a8371f67c31178bbc1311850b05/lockfile/src/spdx.rs#L161-L165